packages/apps/human-app/frontend/src/api/fetcher.ts

```diff
@@ -71,6 +71,46 @@ export function createFetcher(defaultFetcherConfig?: {
   options?: RequestInit | (() => RequestInit);
   baseUrl: FetcherUrl | (() => FetcherUrl);
 }) {
+  let isRefreshing = false;
+  let refreshPromise: Promise<SignInSuccessResponse | undefined> | null = null;
+
+  async function refreshToken(): Promise<
+    { access_token: string; refresh_token: string } | null | undefined
+  > {
+    if (!isRefreshing) {
```

Comment:
You don't need both `isRefreshing` and promise ref, just promise ref is enough.

Let's refactor this function to something more clean, i.e. moving anonymous function with actual logic to separate function and keep `refreshToken` only to call it and handle "single running version" logic.

You can find [example in the fiddle](https://jsfiddle.net/dmitry_hmt_ai/dw056keq/16/)

Reply:
Sure!